lib/po_libc_wrappers.c

```diff
@@ -107,7 +108,28 @@ open(const char *path, int flags, ...)
 	mode = va_arg(args, int);
 	rel = find_relative(path, NULL);
 
-	return openat(rel.dirfd, rel.relative_path, flags, mode);
+	// If the file is already opened, no need of relative opening!
+	if( strcmp(rel.relative_path,".") == 0 )
+		return dup(rel.dirfd);
+	else
+		return openat(rel.dirfd, rel.relative_path, flags, mode);	
+}
+
+/**
+ * Capability-safe wrapper around the `open(2)` system call.
+ *
+ * `open(2)` will behave just like `_open(2)` if the varargs are unpacked and
+ *  passed.
+ */
+int
+open(const char *path, int flags, ...)
+{
+	va_list args;
+	int mode;
+
+	va_start(args, flags);
+	mode = va_arg(args, int);
+    return _open(path, flags, mode);
```

Comment:
Good: let's just fix the whitespace issue here (tab, not spaces) and then this bit will be ready to go.

Reply:
That's assuming that it passes all of the tests... does it?